### PR TITLE
Feature/dash cenc hls fair play example

### PR DESCRIPTION
**Pull request by dominic-miglar**

### CODE REVIEW

File: examples/encoding/02_simple_encoding_dash_cenc_hls_fairplay.js [hunk 1]
```diff
@@ -0,0 +1,655 @@
...
+        codecConfigurationH264At544p,
+        codecConfigurationH264At272p,
+        codecConfigurationAAC
+      ]) => {
+      console.log('Successfully created input, output and codec configurations.');
+      const dashManifestCreationPromise = createDashManifestWithPeriodAndAdaptationSets(output.id);
+      const hlsManifestCreationPromise = createHlsManifest(output.id);
+      Promise.all([dashManifestCreationPromise, hlsManifestCreationPromise]).then(([dashManifest, hlsManifest]) => {
+        console.log('Successfully created DASH and HLS Manifest Resources');
+        [createdDashManifest, createdDashManifestPeriod, createdDashManifestAudioAdaptationSet, createdDashManifestVideoAdaptationSet] = dashManifest;
```
Comment (lukaskroepfl) [inline examples/encoding/02_simple_encoding_dash_cenc_hls_fairplay.js]: sure the `dashManifest` array has always this order?

Comment (dominic-miglar) [inline examples/encoding/02_simple_encoding_dash_cenc_hls_fairplay.js]: yes.

Comment (Tigraine) [inline examples/encoding/02_simple_encoding_dash_cenc_hls_fairplay.js]: In promises the order you add them to the `all([])` array defines the order you get the promise `.then` parameter array @lukaskroepfl

File: examples/encoding/02_simple_encoding_dash_cenc_hls_fairplay.js [hunk 2]
```diff
@@ -0,0 +1,655 @@
...
+            adaptationSet: createdDashManifestVideoAdaptationSet,
+            outputPathDiscriminator: 'video/544'
+          },          {
+            codecConfiguration: codecConfigurationH264At272p,
+            adaptationSet: createdDashManifestVideoAdaptationSet,
+            outputPathDiscriminator: 'video/272'
+          }
+        ];
+
+        const promiseMap = Promise.map(streamDefinitions, (streamDefinition) => {
```
Comment (Tigraine) [inline examples/encoding/02_simple_encoding_dash_cenc_hls_fairplay.js]: Watch out as Promise.map is a bluebird function that is not available in other environments. it's fine but we should document this dependency on bluebird.js

File: examples/encoding/02_simple_encoding_dash_cenc_hls_fairplay.js [hunk 3]
```diff
@@ -0,0 +1,655 @@
...
+          console.error('An error occured: ', error);
+          reject(error);
+        });
+      });
+    });
+  });
+};
+
+const createInput = () => {
+  return new Promise((resolve, reject) => {
```
Comment (Tigraine) [inline examples/encoding/02_simple_encoding_dash_cenc_hls_fairplay.js]: You can cut down the number of promises we throw around if we just do the following:

```
return bitmovin.encoding.inputs.http.create(httpInput).then((createdInput) => {
  console.log(...);
  return createdInput;
}).catch((error) => {
  console.log(...);
  return error;
});
```

Thus saving 2 lines on each promise that is only doing logging..

File: examples/encoding/02_simple_encoding_dash_cenc_hls_fairplay.js [hunk 4]
```diff
@@ -0,0 +1,655 @@
...
+      name: 'My awesome S3 Output',
+      description: 'Little description for my awesome S3 Output',
+      accessKey: OUTPUT_S3_ACCESS_KEY,
+      secretKey: OUTPUT_S3_SECRET_KEY,
+      bucketName: OUTPUT_S3_BUCKET_NAME
+    };
+    bitmovin.encoding.outputs.s3.create(s3Output).then((createdOutput) => {
+      console.log('Successfully created S3 Output.', createdOutput);
+      resolve(createdOutput);
+    }).catch((error) => {
```
Comment (Tigraine) [inline examples/encoding/02_simple_encoding_dash_cenc_hls_fairplay.js]: We could cut down on all the error handling with a function like this:

```
const logErrorPromise = (errorMessage) => {
	return (value) => {
  	console.log(errorMessage, value);
    return value;
  };
};
```

Then we would rewrite this as 
`.catch(logErrorPromise('nable to create S3 Output.'));`

Try it  here: https://jsfiddle.net/39wryt4b/

File: examples/encoding/02_simple_encoding_dash_cenc_hls_fairplay.js (hunk 2 above)
Comment (dominic-miglar) [inline examples/encoding/02_simple_encoding_dash_cenc_hls_fairplay.js]: everyone can see that the bluebird Promise lib is imported at the top of the file.

**Pull request merged**